### ISSUE TITLE
Rule S3903: False Positive when using top level statments and partial Program

### ISSUE THREAD

**Issue by dnperfors** [SonarSource/sonar-dotnet#5660]
### Description

When using top level statements in C# and after specifying a partial Program class, [S3903](https://sonarsource.github.io/rspec/#/rspec/S3903/csharp) is emitted.

### Repro steps

The following code will generate the warning:
```
Console.Write("Hello World");

[ExcludeFromCodeCoverage]
public partial class Program { }
```

### Expected behavior

The S3903 warning is not emitted.

### Actual behavior

The S3903 warning is emitted

### Known workarounds

Other than not using top level statements or suppressing this specific message, there is no sensible work around.

### Related information

* C#/VB.NET Plugins version 8.36.1.44192 [C# Code Quality and Security]
* Visual Studio version 2022 17.1.2
* MSBuild / dotnet version 6.0

**Comment by costin-zaharia-sonarsource** [comment]
Hi @dnperfors,

I'm trying to understand the reasoning behind it. Why do you consider this a false positive? 

From my point of view, the class `Program` from the above snippet is similar to any other class defined as a top-level statement. If a namespace is not provided, it will be added by default to the root namespace.

**Comment by dnperfors** [comment]
When you put the partial `Program` class in a seperate namespace, it will be a separate class from the one that is generated by the compiler. This makes it impossible to add attributes to the `Program` class and to use it in integration tests.
Besides that, a similar warning from .NET is also ignoring the partial `Program` class (for teh same reason (https://github.com/dotnet/roslyn-analyzers/issues/5769)

**Comment by costin-zaharia-sonarsource** [comment]
Perfect. Thanks, @dnperfors, for the clarification.
